main.tf

```
@@ -1 +1,445 @@
+    }
+  )
+}
+
+###
+### DATA SOURCES
+###
+
+data "aws_partition" "current" {
+}
+
+data "aws_caller_identity" "current" {
+}
+
+data "aws_region" "current" {
+}
+
+data "aws_ami" "this" {
+  most_recent = "true"
+
+  owners     = var.ami_owners
+  name_regex = var.ami_name_regex
+  filter {
+    name   = "name"
+    values = var.ami_name_filters
+  }
+}
+
+data "aws_subnet" "lb" {
+  count = length(var.lb_subnet_ids)
+
+  id = var.lb_subnet_ids[count.index]
+}
+
+data "archive_file" "salt" {
+  type        = "zip"
+  source_dir  = join("/", [path.module, "salt"])
+  output_path = join("/", [local.archive_dir_path, local.archive_file_name])
+}
+
+data "archive_file" "configs" {
+  count       = var.vault_configs_path == null ? 0 : 1
+  type        = "zip"
+  source_dir  = var.vault_configs_path
+  output_path = join("/", [local.archive_dir_path, local.configs_file_name])
+}
+
+data "template_file" "appscript" {
+  template = file(join("/", [local.appscript_dir_path, local.appscript_file_name]))
+
+  vars = {
+    salt_content_archive = local.s3_salt_vault_content
+
+    salt_grains_json = join("", ["'", jsonencode({
+      api_port       = var.api_port
+      cluster_port   = var.cluster_port
+      dynamodb_table = local.dynamodb_table
+      inbound_cidrs  = concat(var.inbound_cidrs, local.default_inbound_cdirs)
+      kms_key_id     = local.kms_key_id
+      logs_path      = local.logs_path
+      region         = data.aws_region.current.name
+      ssm_path       = local.ssm_root_path
+      version        = var.vault_version
+    }), "'"])
+  }
+}
+
+# Manage S3 bucket module
+module "s3_bucket" {
+  source  = "terraform-aws-modules/s3-bucket/aws"
+  version = "0.0.1"
+
+  bucket = local.s3_bucket_name
+}
+
+
+resource "aws_s3_bucket_policy" "this" {
+  bucket = module.s3_bucket.id
+  policy = templatefile("${path.module}/policies/bucket_policy.json", { bucket_arn = module.s3_bucket.arn })
+}
+
+# Manage IAM module
+module "iam" {
+  source = "./modules/iam"
+
+  role_name = local.role_name
+  policy_vars = {
+    bucket_name    = module.s3_bucket.id
+    dynamodb_table = local.dynamodb_table
+    kms_key_id     = local.kms_key_id
+    stack_name     = var.name
+    ssm_path       = local.ssm_root_path
+  }
+}
+
+# Generate a random id for each deployment
+resource "random_string" "this" {
+  length  = 8
+  special = false
+  upper   = false
+}
+
+# Manage archive and appscript files
+resource "aws_s3_bucket_object" "salt_zip" {
+  bucket = module.s3_bucket.id
+  key    = local.archive_file_name
+  source = join("/", [local.archive_dir_path, local.archive_file_name])
+  etag   = data.archive_file.salt.output_md5
+}
+
+resource "aws_s3_bucket_object" "configs_zip" {
+  count  = var.vault_configs_path == null ? 0 : 1
+  bucket = module.s3_bucket.id
+  key    = local.configs_file_name
+  source = join("/", [local.archive_dir_path, local.configs_file_name])
+  etag   = data.archive_file.configs[count.index].output_md5
```

we're only dealing with one of something, no need to use count.index

```suggestion
  etag   = data.archive_file.configs[0].output_md5
```